e2e/react-start/basic-cloudflare/tests/app.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { expect } from '@playwright/test'
+import { test } from '@tanstack/router-e2e-utils'
+
+test('returns the correct user agent', async ({ page }) => {
+  await page.goto('/')
+
+  await expect(page.locator('p').nth(0)).toHaveText(
+    'Running in Cloudflare-Workers',
+  )
+})
+
+test('returns the correct value from a Cloudflare binding', async ({
+  page,
+}) => {
+  await page.goto('/')
+
+  await expect(page.locator('p').nth(1)).toHaveText('Hello from Cloudflare')
```

Comment:
please use data-testid instead of locator to make those tests simpler and more robust to changes